### ISSUE TITLE
Question on the MLP for supporting sentences

### ISSUE THREAD

**Issue by allanj** [yuwfan/HGN#6]
I can see that you are using an MLP to produce one single score to supporting sentences. The dimension of MLP is from `2*hidden_size` to `1`. Then the code, concat a '0' to make the last dimension become 2.

https://github.com/yuwfan/HGN/blob/master/models/layers.py#L284-L297
```python
        gat_logit = self.sent_mlp(graph_state[:, :1+max_para_num+max_sent_num, :]) # N x max_sent x 1
        para_logit = gat_logit[:, 1:1+max_para_num, :].contiguous()
        sent_logit = gat_logit[:, 1+max_para_num:, :].contiguous()

        .....
        sent_logits_aux = Variable(sent_logit.data.new(sent_logit.size(0), sent_logit.size(1), 1).zero_())
        sent_prediction = torch.cat([sent_logits_aux, sent_logit], dim=-1).contiguous()
```

I was wondering why it isn't from `2*hidden_size` to `2`. Isn't it more natural for binary classification?

**Comment by yuwfan** [comment]
Thanks Allan for pointing out this. I think you could choose either way and the differences should be minor. BTW since our repo is based on DFGN, we directly followed their implementation (https://github.com/woshiyyya/DFGN-pytorch/blob/569bfdd67d8e54bb244339965a9268fb64806014/DFGN/model/layers.py#L425).

**Comment by allanj** [comment]
Thanks for answering that. Now I think I'm clear.

Another question just think of in my mind is that:

We have an MTL loss in the paper. I wonder if you have also tried to train each loss/task separately.

In my task, (probably/partially because my dataset size is small), train supporting fact loss and classification loss together could be harmful to both tasks or to the supporting fact extraction task.

**Comment by yuwfan** [comment]
We haven't ablated answer extraction and supporting loss since they are two essential tasks in HotpotQA. For paragraph and entity loss, their improvement is quite marginal, but no harm to both tasks.
